### PR TITLE
Use extended Sqlite error codes

### PR DESCRIPTION
**Pull request by AlexanderTaeschner**
Addresses #404

### CODE REVIEW

**Comment by bricelam** [comment]
This PR raises another concern: If the app calls `sqlite3_extended_result_codes()` does it break us? If so, we should file a new issue to be resilient to this.

**Comment by AlexanderTaeschner** [comment]
Changed the PR by adding the property to SqliteException.

If someone calls ` sqlite3_extended_result_codes()` it breaks the check https://github.com/aspnet/Microsoft.Data.Sqlite/blob/791993269cfbf97ec3fbdb5139d860ba61ef7396/src/Microsoft.Data.Sqlite.Core/SqliteCommand.cs#L355
we could add `|| rc == raw.SQLITE_LOCKED_SHAREDCACHE` to be safe.

**Comment by bricelam** [comment]
Yeah, seems like we should update the check

File: src/Microsoft.Data.Sqlite.Core/SqliteException.cs [hunk 1]
```diff
@@ -49,8 +69,11 @@ public static void ThrowExceptionForRC(int rc, sqlite3 db)
             var message = db == null || db.ptr == IntPtr.Zero
                 ? raw.sqlite3_errstr(rc) + " " + Resources.DefaultNativeError
                 : raw.sqlite3_errmsg(db);
+            var extendedErrorCode = db == null || db.ptr == IntPtr.Zero
+                ? rc
+                : raw.sqlite3_extended_errcode(db);
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite.Core/SqliteException.cs]: DRY:
``` csharp
string message;
int extendedErrorCode;
if (db == null || db.ptr == IntPtr.Zero)
{
    extendedErrorCode = rc;
    message = raw.sqlite3_errstr(rc) + " " + Resources.DefaultNativeError;
}
else
{
    message  = raw.sqlite3_errmsg(db);
    extendedErrorCode  = raw.sqlite3_extended_errcode(db);
}
```

Comment (AlexanderTaeschner) [reply src/Microsoft.Data.Sqlite.Core/SqliteException.cs]: Rebased on top of dev and corrected the DRY issue.

**Pull request merged**